cloudflare.go

```diff
@@ -90,11 +90,23 @@ func (api *API) ZoneIDByName(zoneName string) (string, error) {
 	if err != nil {
 		return "", errors.Wrap(err, "ListZones command failed")
 	}
+
+	if len(res) > 1 && api.OrganizationID == "" {
+		return "", errors.New("ambiguous zone name used without an account ID")
+	}
+
 	for _, zone := range res {
-		if zone.Name == zoneName {
-			return zone.ID, nil
+		if api.OrganizationID != "" {
```

Comment:
Is the OrganizationID attribute of an API object ever set?  When we instantiate an API in cmd/flarectl/flarectl.go line 31, we provide only two parameters: API key and API e-mail address.  An organization ID will never be used in this case.

```go
api, err = cloudflare.New(os.Getenv("CF_API_KEY"), os.Getenv("CF_API_EMAIL"))
```

Further, multiple matches for a zone lookup by zone name is indication of a problem regardless of whether or not our API client has an OrganizationID attribute.

Reply:
Yes, `OrganizationID` is used in https://github.com/terraform-providers/terraform-provider-cloudflare, it may also be used by other third-party projects as SDK.

Possibility of returning same zone twice is because our state machine allows same zone name be signed up to many accounts. They will be separate instances (different zone IDs), but only one of them will be active at a time.

This is allowed to enable zone migration between accounts without need for downtime.